storage/client_range_test.go

```diff
@@ -150,3 +155,156 @@ func TestRejectFutureCommand(t *testing.T) {
 		t.Errorf("expected 15, got %v", v)
 	}
 }
+
+// Test a case where a put operation of an older timestamp comes after
+// a put operation of a newer timestamp.
+//
+// 1) Txn executes a put operation with time T.
+// 2) Before the txn is committed, another client sends a get
+//    operation with time T+100. This triggers the txn restart.
+// 3) The client sends another get operation with time T+200. The
+//    write intent is resolved (and the txn timestamp is pushed to
```

Comment:
why is the intent resolved? The transaction is still running. It will only push its timestamp, right?